lib/hammer_cli_foreman/smart_class_parameter.rb

```diff
@@ -127,9 +127,9 @@ class UpdateCommand < HammerCLIForeman::UpdateCommand
       end
     end
 
-    class AddOverrideValueCommand < HammerCLIForeman::CreateCommand
+    class AddMatcher < HammerCLIForeman::CreateCommand
```

Comment:
@rahulbajaj0509 ,

@tstrachota mentioned on [issue-12999](http://projects.theforeman.org/issues/12999) - 

`Please note that we need to keep original 'add-override-value' command in place for backwards compatibility. It can probably be hidden alias to the new command with a deprecation warning.`

Please check on this.

Reply:
So @tstrachota ,
you meant : make a new sub-command with the name `add-matcher` and make the command `add-override-value` hidden, marking it with a  deprecated warning. Am i correct ?

Reply:
@rahulbajaj0509 correct. Help should list only `add-matcher` but using `add-override-value` should still be possible. It will only print a warning to output.